aten/src/ATen/NestedTensorImpl.cpp

```diff
@@ -230,12 +230,25 @@ NestedTensorImpl::NestedTensorImpl(
       nested_sizes_(std::move(nested_sizes)),
       nested_strides_(std::move(nested_strides)),
       storage_offsets_(std::move(storage_offsets)),
-      opt_sizes_(construct_opt_sizes(nested_sizes_)) {
+      opt_sizes_(c10::nullopt) {
   validate_nested_tensor_metadata(nested_sizes_, nested_strides_, storage_offsets_);
   refresh_dim();
   set_custom_sizes_strides(c10::TensorImpl::SizesStridesPolicy::CustomSizes);
 }
 
+c10::optional<int64_t> NestedTensorImpl::opt_size(int64_t d) const {
+  if (C10_UNLIKELY(!opt_sizes_.has_value())) {
+    // Maintain const-ness for this method even though we're technically mutating.
+    *(const_cast<c10::optional<std::vector<int64_t>>*>(&opt_sizes_)) = c10::make_optional(
```

Comment:
I can't comment on whether this is a good idea or not. It doesn't look like a good idea?

Reply:
Perhaps not, but the const-ness is required by `size()`, and that follows the `TensorImpl` interface so it can't change

Reply:
After reading this (stack overflow)[https://stackoverflow.com/questions/3484233/const-method-that-modifies-this-without-const-cast] I am convinced this is okay

Reply:
I think I should be using the [`mutable` keyword](https://www.geeksforgeeks.org/c-mutable-keyword/) for `opt_sizes_`.

Reply:
!00%
<img width="879" alt="Screenshot 2023-03-29 at 12 54 51 PM" src="https://user-images.githubusercontent.com/32754868/228652572-b5b9e4ee-1a04-4473-9bf2-c53dbbec1da0.png">